stockfish/models.py

```diff
@@ -998,28 +1008,124 @@ def will_move_be_a_capture(self, move_value: str) -> Capture:
+                "full": 0,
+                "major": 0,
+                "minor": 0,
+                "patch": "",
+                "sha": "",
+                "is_dev_build": False,
+                "text": version_text,
+            }
+
+            # check if version is a development build, eg. dev-20221219-61ea1534
+            if self._version["text"].startswith("dev-"):
+                self._version["is_dev_build"] = True
+
+                # parse patch and sha from dev version text
+                self._version["patch"] = self._version["text"].split("-")[1]
+                self._version["sha"] = self._version["text"].split("-")[2]
+
+                # get major.minor version as text from build date
+                build_date = self._version["text"].split("-")[1]
+                date_string = f"{int(build_date[:4])}-{int(build_date[4:6]):02d}-{int(build_date[6:8]):02d}"
+                self._version["text"] = self._get_stockfish_version_from_build_date(
+                    date_string
+                )
+
+            # check if version is a development build, eg. 280322
+            if len(self._version["text"]) == 6:
+                self._version["is_dev_build"] = True
+
+                # parse version number from DDMMYY
+                self._version["patch"] = self._version["text"]
+
+                # parse build date from dev version text
+                build_date = self._version["text"]
+                date_string = f"20{build_date[4:6]}-{build_date[2:4]}-{build_date[0:2]}"
+                self._version["text"] = self._get_stockfish_version_from_build_date(
+                    date_string
+                )
+
+            # parse version number for all versions
+            self._version["major"] = int(self._version["text"].split(".")[0])
+            try:
+                self._version["minor"] = int(self._version["text"].split(".")[1])
+            except IndexError:
+                self._version["minor"] = 0
+            self._version["full"] = self._version["major"] + self._version["minor"] / 10
+        except Exception as e:
+            raise Exception(
+                "Unable to parse Stockfish version. You may be using an unsupported version of Stockfish."
+            )
+
+    def _get_stockfish_version_from_build_date(
+        self, date_string: str = ""
+    ) -> Optional[str]:
+        # Convert date string to datetime object
+        date_object = datetime.strptime(date_string, "%Y-%m-%d")
+
+        # Convert release date strings to datetime objects
+        releases_datetime = {
+            key: datetime.strptime(value, "%Y-%m-%d")
+            for key, value in self._releases.items()
+        }
+
+        # Find the key for the given date
+        key_for_date = None
+        for key, value in releases_datetime.items():
+            if value <= date_object:
+                if key_for_date is None or value > releases_datetime[key_for_date]:
+                    key_for_date = key
+
```

Comment:
```suggestion
        if key_for_date is None:
            raise Exception("There was a problem with finding the release associated with the engine publish date.")
```
It might be nice to get a more detailed error message when an error occurs here. The above is just a suggestion, feel free to use a different wording.

Reply:
Do we want to throw an exception here, though? Even if it's not possible to determined dev version, the program will very likely work as expected?

Reply:
Well, if `key_for_date` is None an exception is raised [here](https://github.com/knutole/stockfish/blob/0fa500b49fbdc996994054bce52a2e3cbf3290de/stockfish/models.py#LL1106) anyway, because the value of `self._version["text"]` is set to None and cannot [be split](https://github.com/knutole/stockfish/blob/0fa500b49fbdc996994054bce52a2e3cbf3290de/stockfish/models.py#L1098). If it raises an exception anyway and you raise one before for this special case, it would look like this:
> [...]
> Exception: There was a problem with finding the release associated with the engine publish date.
> 
> During handling of the above exception, another exception occurred:
>  [...]
> Exception: Unable to parse Stockfish version. You may be using an unsupported version of Stockfish.


I'm completely fine with it not raising an exception at all, but my idea was that if it raises one it should pinpoint the exact location of the problem.

Reply:
Yes, I see now, thanks! :) I've added the exception 👍